### PR TITLE
CSHARP-806 Drop netstandard1.5 and bump net45 to net452

### PR DESCRIPTION
**Pull request by joao-r-reis**
Notable changes are the addition of json converters to graph and geometry types. Instead of these converters we could bump newtonsoft from 9.0.1 to 10.0.3 (where newtonsoft supports `ISerializable` in .net core) but I figured this could be annoying for OSS users (or even dse users that don't use graph) so the json converter does the trick without having to bump the version.

I enabled the graph serialization unit tests in .net core and added an `Assert` for json deserialize (existing tests only tested serialize). Also added these tests for GraphNode with GraphSON2 (previously had a test with GraphSON1 only).

### CODE REVIEW

**Comment by joao-r-reis** [comment]
Oh and I haven't changed the matrix yet because we have a couple of unit tests that depend on the runtime. I'm thinking of leaving the matrix unchanged for appveyor and travis but remove `netcoreapp2.0` (which targeted driver's `netstandard1.5` before this PR) integration tests from jenkins.

I'll also take this opportunity to remove `DSE` + `Windows` tests from jenkins but keep `OSS` + `Windows`

File: src/Cassandra.IntegrationTests/Cassandra.IntegrationTests.csproj [hunk 1]
```diff
@@ -12,12 +12,18 @@
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
+  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '^net\d'))">
+    <DefineConstants>$(DefineConstants);NETFRAMEWORK</DefineConstants>
+  </PropertyGroup>
+  <PropertyGroup Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('$(TargetFramework)', '^netcoreapp\d'))">
```
[the remark this answers is not in the document]
Comment (jorgebay) [reply src/Cassandra.IntegrationTests/Cassandra.IntegrationTests.csproj]: Also `NETFRAMEWORK` and `NETCOREAPP` are more clear 👍

[the remark this answers is not in the document]
Comment (joao-r-reis) [reply src/Cassandra.IntegrationTests/Cassandra.IntegrationTests.csproj]: Yeah starting from a certain dotnet SDK version these are already included but this removes our dependency to a specific version. 

I thought that these were always included so it took me a while to figure out what was causing unit test failures on travis... 😅

File: README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ It also provides additional features for [DataStax Enterprise][dse]:
 - [DSE graph][dse-graph] integration.
 - Serializers for geospatial types which integrate seamlessly with the driver.
 
-The driver supports .NET Framework 4.5+ and .NET Core 1+.
+The driver supports .NET Framework 4.5.2+ and .NET Standard 2.0+.
```
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply README.md]: Yeah looking back I shouldn't have changed this. I'll fix it in a few mins

File: doc/features/cloud/README.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ var session =
 
 DataStax Apollo support on .NET Core requires **.NET Core Runtime 2.1 or later**.
 
-For the remaining platforms supported by the driver, there is no additional requirement.
+On .NET Framework there is no additional requirement.
```
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply doc/features/cloud/README.md]: done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -372,8 +372,8 @@ session.Execute(statement);
 ## Compatibility
 
 - Apache Cassandra versions 2.0 and above.
-- DataStax Enterprise versions 4.5 and above.
-- .NET Framework versions 4.5 and above and .NET Core versions 1.0 and above.
+- DataStax Enterprise versions 4.8 and above.
+- .NET Framework versions 4.5.2 and above and .NET Standard versions 2.0 and above.
```
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply README.md]: done

**Pull request merged**